packages/popover/src/Popover/PopoverContent.js

```diff
@@ -0,0 +1,169 @@
+    const keys = {
+      // Target ESC key
+      27: () => {
+        evt.preventDefault()
+        onCloseRequest()
+      }
+    }
+
+    if (keys[evt.keyCode]) {
+      keys[evt.keyCode]()
+    }
+  }
+
+  calcOffset(placement) {
+    // Need to account for the padding added around the popover, this fixes the offset at the start and end position
+    if (RegExp('start*').test(placement)) {
+      return '-16px'
+    } else if (RegExp('end*').test(placement)) {
+      return '16px'
+    } else {
+      return 0
+    }
+  }
+
+  render() {
+    const { onCloseRequest } = this.props
+    const styleProps = {
+      theme: this.props.theme,
+      background: this.props.bg,
+      borderColor: this.props.borderColor,
+      zIndex: this.props.zIndex,
+      width: this.props.width
+    }
+
+    return ReactDOM.createPortal(
+      <React.Fragment>
+        <Popper
+          positionFixed={true}
+          modifiers={{
+            ...DEFAULTS_MODIFIERS,
+            offset: {
+              offset: this.calcOffset(this.props.placement)
+            }
+          }}
+          {...this.props}
+        >
+          {({ placement, ref, style, arrowProps }) => (
+            // Need to be a native element, because of ref forwarding limitations with DS functional components
+            <PopperGuide
+              /*
+               * NOTE: InnerRef has been depracted in V4 of styled components. Make sure to change this prop once we upgrade to styled components v4
+               * https://www.styled-components.com/docs/api#deprecated-innerref-prop
+               */
+              className={this.props.className}
+              innerRef={ref}
+              style={style}
+              data-placement={placement}
+              aria-label={this.props.ariaLabel}
+              {...styleProps}
+              role="dialog"
+              aria-describedby={`dialog-description-${this.props.idx}`}
+            >
+              <ContentContainer
+                innerRef={this.props.contentRef}
+                {...styleProps}
+                tabIndex="-1"
+              >
+                <Box id={`popover-description-${this.props.idx}`}>
+                  {this.props.renderContent({
+                    handleClose: onCloseRequest
+                  })}
+                </Box>
+              </ContentContainer>
+              <PopoverArrow
+                arrowProps={arrowProps}
+                placement={placement}
+                background={styleProps.background}
+                borderColor={styleProps.borderColor}
+              />
+            </PopperGuide>
+          )}
+        </Popper>
+        <Overlay
+          handleClick={onCloseRequest}
+          zIndex={this.props.zIndex - 1}
+          overlayOpacity={this.props.overlayOpacity}
+        />
+      </React.Fragment>,
+      // Append each instance of the Popover as portal directly to the body
+      document.querySelector('body')
+    )
+  }
+}
+
+const PopperGuide = styled.div`
```

Comment:
Before, I was having issue passing refs to DS, so that's why I decided to use a native element. But then I found out that if we pass the ref as `innerRef` it works with DS and just forgot to update this part of the code... oops :)